src/bstring.ml

```diff
@@ -38,39 +45,78 @@ let compare = String.compare
 
 let hash s = Hashtbl.hash s
 
-let init = String.init
+let make = String.make
 
+let makeWithInit = String.init
+
+let substring s ~from ~length = String.sub s from length
+
+let concat = String.concat
+
+let blit = String.blit
+
+let uncapitalize = String.uncapitalize
+let capitalize = String.capitalize
+let lowercase = String.lowercase
+let uppercase = String.uppercase
+let escaped = String.escaped
+let trim = String.trim
 
 (*let init n f =
   let buf = Bytes.init n f in
   Bytes.unsafe_to_string buf*)
 
+external get : string -> int -> char = "%string_safe_get"
+
 let length = String.length
 
-let rev s =
+let forEach = String.iter
+let map = String.map
+let forEachWithIndex = String.iteri
+let mapWithIndex = String.mapi
+
+let reverse s =
   let n = length s in
-  init n (fun i -> s.[n-i-1])
+  makeWithInit n (fun i -> s.[n-i-1])
 
-let rec _to_list s acc i len =
+let rec _toList s acc i len =
```

Comment:
I suggest we expose this API. I found it quite useful, also would parallel `toList` in `Barray`